src/library.js

```diff
@@ -569,7 +569,7 @@ mergeInto(LibraryManager.library, {
 
   // TODO: Initialize these to defaults on startup from system settings.
   // Note: glibc has one fewer underscore for all of these. Also used in other related functions (timegm)
-  _tzset_js__deps: ['$stringToNewUTF8'],
+  _tzset_js__deps: ['$stringToNewUTF8', 'malloc'],
```

Comment:
Where is malloc used? I don't seem to see it in this function.

Also, could this change be split out?

Reply:
The issue here is `stringToNewUTF8`, which instead of directly depending on `malloc`, uses `makeMalloc` to essentially make the dependency optional and instead fail and runtime if malloc is missing.

So its up to each caller of `stringToNewUTF8` to decide it wants to explictly depend on `malloc` or not.   This is a pre-existing behaviour, that can explain a lot of the additional malloc deps that are part of this PR.

You can see that this dependency was also present if the existing deps_info.py (see ` 'tzset': ['malloc'],` in that file).

Do you think its worth splitting of a PR that basically mirrors the deps from deps_info.py into the `__deps`?   Sounds reasonable to me..

Reply:
Splitting out sounds good. But I wonder if we can make deps like this point only to `stringToNewUTF8`, and let that function handle adding `malloc` as needed? That is, I'm not sure why one step back in the chain needs to do it, in multiple places?

Reply:
Done.  This landed in #19160